### PR TITLE
KAFKA-16975 The error message of creating `__cluster_metadata` should NOT be "Authorization failed"

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-16975
The error message of creating "__cluster_metadata" by admin is "Authorization failed.". That could confuse users that it implies you "can" create it in using root. While creating this topic is prohibited. This pr corrected the error code and error message in KafkaApis when user creating __cluster_metadata topic.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/ControllerApis.scala [hunk 1]
```diff
@@ -435,7 +435,12 @@ class ControllerApis(
           setErrorMessage("Duplicate topic name."))
       }
       topicNames.forEach { name =>
-        if (!authorizedTopicNames.contains(name)) {
+        if (name == Topic.CLUSTER_METADATA_TOPIC_NAME) {
+          response.topics().add(new CreatableTopicResult().
+            setName(name).
+            setErrorCode(INVALID_REQUEST.code).
+            setErrorMessage("Internal topic creation is prohibited."))
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/ControllerApis.scala]: Could it include the topic name?

File: core/src/test/scala/unit/kafka/server/ControllerApisTest.scala [hunk 2]
```diff
@@ -680,7 +682,10 @@ class ControllerApisTest {
         setTopicConfigErrorCode(TOPIC_AUTHORIZATION_FAILED.code()),
       new CreatableTopicResult().setName("quux").
         setErrorCode(TOPIC_AUTHORIZATION_FAILED.code()).
-        setErrorMessage("Authorization failed."))
+        setErrorMessage("Authorization failed."),
+      new CreatableTopicResult().setName(Topic.CLUSTER_METADATA_TOPIC_NAME).
+        setErrorCode(INVALID_REQUEST.code()).
+        setErrorMessage("Internal topic creation is prohibited."))
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/ControllerApisTest.scala]: Could it include the topic name?

**Review by chia7712 — COMMENTED** [review]
@brandboat thanks for this patch.

@jsancio Could you please take a look? not sure why #12885 decided to return  Authorization error instead of bad request. Please feel free to correct me. thanks

**Comment by jsancio** [comment]
> @jsancio Could you please take a look? not sure why #12885 decided to return Authorization error instead of bad request. Please feel free to correct me. thanks

I don't have a strong argument for returning an authorization error vs an invalid request error as long as the UX makes sense to the user.

@brandboat take a look at this PR https://github.com/apache/kafka/pull/12885 and make sure that you also remove any code that I added to return authorization errors when creating the __cluster_metadata topic. This PR returns an invalid request error but leaves being code that is not needed anymore.

**Comment by brandboat** [comment]
> @brandboat take a look at this PR https://github.com/apache/kafka/pull/12885 and make sure that you also remove any code that I added to return authorization errors when creating the __cluster_metadata topic. This PR returns an invalid request error but leaves being code that is not needed anymore.

Thank you @jsancio. I've checked the PR you mentioned and didn't find any unused items left behind, perhaps I overlooked something...

**Review by chia7712 — COMMENTED** [review]
@brandboat This PR returns the error code with obvious message, so it seems we don't need to log it. Hence, could you please remove following log?

https://github.com/apache/kafka/blob/ec9dabf86fd0ddd006cf26cd5d4dc967fdd6826f/core/src/main/scala/kafka/server/ControllerApis.scala#L411

https://github.com/apache/kafka/blob/ec9dabf86fd0ddd006cf26cd5d4dc967fdd6826f/core/src/main/scala/kafka/server/KafkaApis.scala#L2022

**Comment by brandboat** [comment]
> @brandboat This PR returns the error code with obvious message, so it seems we don't need to log it. Hence, could you please remove following log?

Make sense to me, I removed those logs in the latest commit. Thank you.

**Comment by chia7712** [comment]
@brandboat Could you please rebase code to trigger QA again? Those failed tests pass on my local, but I'd like to run them again on our QA.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**